src/main.ts

```diff
@@ -1,16 +1,25 @@
-import App from './app/App.vue'
-import { TOKENS, get } from '@/services'
+import { TOKENS as $, services as production } from '@/services/production'
+import { build } from '@/services/utils'
 
 async function mountVueApplication() {
-  const env = get(TOKENS.env)
-  const kumaApi = get(TOKENS.api)
-  // During development setBaseUrl also optionally installs MSW mocking via MockKumaApi
-  kumaApi.setBaseUrl(env('KUMA_API_URL'))
-
-  const app = await get(TOKENS.app)(App)
+  const get = build(
+    // production service container configuration
+    production($),
+    // any DEV-time only service container configuration
+    import.meta.env.DEV
+      ? await (async () => {
+        const dev = await import('@/services/development')
+        return dev.services({
+          ...$,
+          ...dev.TOKENS,
+        })
+      })()
+      : [],
+  )
```

Comment:
I'm pretty sure this gives us a reliable "don't add the code in here to a production build" wrapper, if someone could just give that a second set of eyes that'd be much appreciated.

It's done in this way, as a function wrapper rather than a single line, so we can add more to it in other contexts. It's a little wieldy so an improvement on this that achieves the same would be good also, no matter if not though its code that will hardly ever be changed and is only in one place.

(this was one usecase where previously we were relying on vite aliasing)

Reply:
This should work. The combination of compiler-analizeable condition (i.e. the ternary involving `import.meta.env.DEV`) and a dynamic import should remove any trace of the development file in a non-dev build.